frontend/src/metabase/admin/permissions/utils/graph/data-permissions.ts

```diff
@@ -188,6 +188,30 @@ export const getFieldsPermission = (
   }
 };
 
+export function hasPermissionValueInGraph(
+  permissions: GroupsPermissions,
+  permissionValue: DataPermissionValue,
+): boolean {
+  function _hasPermissionValueInGraph(permissionsGraphSection: any) {
+    for (const key in permissionsGraphSection) {
+      if (permissionsGraphSection[key] === permissionValue) {
+        return true;
+      } else if (
+        typeof permissionsGraphSection[key] === "object" &&
+        hasPermissionValueInGraph(permissionsGraphSection[key], permissionValue)
+      ) {
+        return true;
+      } else {
+        continue;
+      }
+    }
+
+    return false;
+  }
+
+  return _hasPermissionValueInGraph(permissions);
+}
```

Comment:
this makes my brain sweat, but I suppose it's a reasonable method

Reply:
Bonus points for recursion! haha

Reply:
i made the checks named variables to hopefully clear up the confusion.

Reply:
also if y'all could let me know where i can cash in on all these bonus points i've been building up 😏